src/lib/storage/run_length_segment/run_length_encoder.hpp

```diff
@@ -28,6 +28,11 @@ class RunLengthEncoder : public SegmentEncoder<RunLengthEncoder> {
     auto iterable = ValueSegmentIterable<T>{*value_segment};
 
     iterable.with_iterators([&](auto it, auto end) {
+      // Early out for empty segments, code below assumes it to be non-empty
+      if (it == end) {
+        return;
+      }
+
       // Init is_current_null such that it does not equal the first entry
       auto current_value = T{};
```

Comment:
Since you are calling default constructors like this most of the time, for consistency:
```suggestion
      T current_value;
```
Although I prefer `auto current_value = T{}`, see [here why](https://www.fluentcpp.com/2018/09/28/auto-stick-changing-style/).